arbnode/message_pruner.go

```diff
@@ -89,30 +89,31 @@ func (m *MessagePruner) prune(ctx context.Context) time.Duration {
 }
 
 func deleteOldMessageFromDB(endBatchCount uint64, endBatchMetadata BatchMetadata, inboxTrackerDb ethdb.Database, transactionStreamerDb ethdb.Database) {
-	var allPrunedKeys [][]byte
 	prunedKeys, err := deleteFromLastPrunedUptoEndKey(inboxTrackerDb, sequencerBatchMetaPrefix, endBatchCount)
 	if err != nil {
 		log.Error("error deleting batch metadata: %w", err)
 		return
 	}
-	allPrunedKeys = append(allPrunedKeys, prunedKeys...)
+	if len(prunedKeys) > 0 {
+		log.Info("Pruned batches:", "first pruned key", prunedKeys[0], "last pruned key", prunedKeys[len(prunedKeys)-1])
+	}
 
 	prunedKeys, err = deleteFromLastPrunedUptoEndKey(transactionStreamerDb, messagePrefix, uint64(endBatchMetadata.MessageCount))
```

Comment:
There are currently 78 million nitro messages on arbitrum one, which means that this will allocate an array with 78 million elements. I'm a bit worried about overhead there. If each element only takes up 8 bytes, it'd only be half a gigabyte, but there will be more overhead than that because each byte array has its own length and capacity which are probably 8 bytes each, and the go gc probably adds overhead, etc. Could we modify `deleteFromLastPrunedUptoEndKey` to only store the first and last keys?

Reply:
Ah, I should have noticed this while removing logs, done.